### ISSUE TITLE
Bug: OneDrive Business | SharePoint regression in handling uploading new and modified content to OneDrive Business and SharePoint

### ISSUE THREAD

**Issue by abraunegg** [abraunegg/onedrive#3020]
### Describe the bug

**Regression from v2.4.25**
* OneDrive Business | SharePoint not adhering to uploading data via UploadSession resource
* Creates multiple versions post upload (potentially a side effect of the above point)
* These multiple versions count towards a users quota

### Operating System Details

```shell
N/A
```

### Client Installation Method

From Source

### OneDrive Account Type

Business | Office365

### What is your OneDrive Application Version

v2.5.3

### What is your OneDrive Application Configuration

```shell
N/A
```

### What is your 'curl' version

```shell
N/A
```

### Where is your 'sync_dir' located

Local

### What are all your system 'mount points'

```shell
N/A
```

### What are all your local file system partition types

```shell
N/A
```

### How do you use 'onedrive'

N/A

### Steps to reproduce the behaviour

* Upload Microsoft Word Document > 4Mb
  * Should use session upload
* Upload Microsoft Word Document < 4Mb
  * Should use session upload - currently does not

### Complete Verbose Log Output

```shell
N/A
```

### Screenshots

N/A

### Other Log Information or Details

```shell
N/A
```

### Additional context

Investigation post discussion here: https://github.com/abraunegg/onedrive/issues/2803

**Comment by abraunegg** [comment]
#### current master - upload modified file < 4mb - online history before upload
![image](https://github.com/user-attachments/assets/acfce798-2312-4e90-8fb4-5176569e868e)

Initial version created by uploading file via web

#### current master - upload modified file < 4mb - online history post upload
```
Reading configuration file: /home/alex/.config/onedrive/config
Configuration file successfully loaded
Using IPv4 and IPv6 (if configured) for all network operations
Attempting to contact Microsoft OneDrive Login Service
Successfully reached Microsoft OneDrive Login Service
Configuring Global Azure AD Endpoints
Fetching items from the OneDrive API for Drive ID: b!6YStAVcRrkaXaJQ7jj2CWfQyuny4Y1ROgq7SUuWfQT9l3Nzwzg_8Roub8Te9E-XZ .. 
No changes or items that can be applied were discovered while processing the data received from Microsoft OneDrive
Performing a database consistency and integrity check on locally stored data ..................................... 
Changed local items to upload to Microsoft OneDrive: 1
Uploading modified file: file_under_4Mb/no_download_access.docx ... done
WARNING: Online file integrity failure for: file_under_4Mb/no_download_access.docx
To disable the integrity checking of uploaded files use --disable-upload-validation
Scanning the local file system 'OneDrive_100K_Files' for new data to upload ........................... 
Performing a last examination of the most recent online data within Microsoft OneDrive to complete the reconciliation process
Fetching items from the OneDrive API for Drive ID: b!6YStAVcRrkaXaJQ7jj2CWfQyuny4Y1ROgq7SUuWfQT9l3Nzwzg_8Roub8Te9E-XZ . 
Processing 2 applicable changes and items received from Microsoft OneDrive . 

Sync with Microsoft OneDrive is complete
Attempting to perform a database vacuum to optimise database
Database vacuum is complete
```
![image](https://github.com/user-attachments/assets/80f2ed83-7cdf-4b4f-9b9c-764c7bec244b)

2 x versions ... this is NOT correct ........ (BUG)

**Comment by abraunegg** [comment]
@NovaPixell
Regarding your comments and feedback in https://github.com/abraunegg/onedrive/issues/2803 I have created this issue ticket and the following PR below.

What this PR should do is no longer just blindly create a new version of a file when uploaded (new or modified), and aligns to how v2.4.x codebase operated.

Please can you test the PR below.

First install all the require platform dependencies to build the client on your respective platforms. Please read https://github.com/abraunegg/onedrive/blob/master/docs/install.md#building-from-source---high-level-requirements and then follow correctly for your platform.

Once this is done, to clone the PR to resolve your issue, you can use a script like the following:
```
#!/bin/bash

PR=3032

rm -rf ./onedrive-pr${PR}
git clone https://github.com/abraunegg/onedrive.git onedrive-pr${PR}
cd onedrive-pr${PR}
git fetch origin pull/${PR}/head:pr${PR}
git checkout pr${PR}

# Configure and Build
./configure --enable-debug --enable-notifications; make clean; make;
./onedrive --version
```

This script will create a local folder called `onedrive-pr3032` with the PR version.

To run the PR, you need to run the client from the PR build directory:
```
./onedrive <any other options needed>
```
To install the PR, you will need to perform `sudo make install` to install the PR version to your system.

When running the PR, your version should be: `onedrive v2.5.3-21-gacc254f` or greater.

**Comment by abraunegg** [comment]
@NovaPixell

Have you been able to look at this issue and/or the associated PR to resolve the 'version' creation issue ?

**Comment by abraunegg** [comment]
This issue has been automatically locked since there has not been any recent activity after it was closed. Please open a new issue for related bugs.
